src/lib.rs

```diff
@@ -97,91 +107,52 @@ compile_error!(
     "
 );
 
-pub use embedded_hal as hal;
-
-pub use nb;
-pub use nb::block;
-
-pub use embedded_time as time;
-
-#[cfg(feature = "defmt")]
-pub(crate) use defmt::{assert, panic, unreachable, unwrap};
-#[cfg(feature = "defmt")]
-mod macros {
-    /// Wrapper function for `.exepct()`
-    ///
-    /// Uses [`defmt::unwrap!`] instead, because
-    /// it has the same functionality as `expect()`
-    #[macro_export]
-    macro_rules! expect {
-        ($l:expr, $s:tt) => {
-            defmt::unwrap!($l, $s)
-        };
-    }
-}
+use cfg_if::cfg_if;
 
-#[cfg(not(feature = "defmt"))]
-pub(crate) use core::{assert, panic, unreachable};
-#[cfg(not(feature = "defmt"))]
-mod macros {
-    /// Wrapper macro for `.unwrap()`
-    ///
-    /// Uses core function, when defmt is not active
-    #[macro_export]
-    macro_rules! unwrap {
-        ($l:expr) => {
-            $l.unwrap()
-        };
-    }
+cfg_if! {
+    if #[cfg(feature = "device-selected")] {
```

Comment:
Because missing `device-selected` does throw a `compile_error!` above already, I wonder if this check is needed at all. 
You could remove it, to save us some indentation :wink:

Reply:
I suggest gating everything on `device-selected` anyway.  This way, if no device is selected, the only error message is the one from `compile_error!` and there aren't millions of "item not found" errors alongside it which clutter the compiler output and make it hard to see the original error.

Reply:
Good point. Than keep it as is :)

Reply:
I found that we should check `device-selected` in `build.rs`, to avoid `thread 'main' panicked at 'internal error: entered unreachable code', build.rs:39:9` when enable `ld` feature without specifying device (`cargo check --features ld`).

Reply:
Either that, or throwing an error, which gives a clear hint to the cause. 

But checking if `device-selected` was set at the before calliong`gen_memory_x` makes sense. Anyway, I would still prefer clearer error-handling instead of `unreachable!` in the `_`-arm in the match cases.